File: tareas_go/commands/agregar.go
package commands

import (
	"fmt"
	"tareas_go/estructuras"
	"time"
)

func (g *GestorTareas) AgregarTarea(titulo, descripcion string) estructuras.Tarea {
	nueva := estructuras.Tarea{
		ID:          g.nextID,
		Titulo:      titulo,
		Descripcion: descripcion,
		Estado:      estructuras.Pendiente,
		Fecha:       time.Now(),
	}

	g.tareas = append(g.tareas, nueva)
	g.nextID++

	g.GuardarTareasJSON()

	fmt.Printf("Tarea agregada: [%d] %s\n", nueva.ID, nueva.Titulo)
	return nueva
}

File: tareas_rust/src/commands/agregar.rs
use crate::commands::persistencia;
use crate::commands::gestor::GestorTareas;
use crate::estructuras::Tarea;

pub fn agregar(g: &mut GestorTareas, titulo: &str, descripcion: &str) -> Result<(), String> {
    let id = g.next_id;
    let nueva = Tarea::nueva(id, titulo, descripcion);
    g.tareas.push(nueva);
    g.next_id += 1;

    persistencia::guardar_tareas_json(&g.tareas)?;
    println!("Tarea agregada: [{}] {}", id, g.tareas.last().unwrap().titulo);
    Ok(())
}
